packages/ui/src/components/va-tabs/VaTabs.vue

```diff
@@ -60,350 +59,388 @@
-    if (this.$props.vertical) {
-      const sliderOffsetY = this.containerRef.clientHeight - tab.$el.offsetTop - tab.$el.clientHeight
-      this.sliderOffsetY = Math.max(sliderOffsetY, 0)
-      this.sliderHeight = tab.$el.clientHeight
-      this.sliderOffsetX = 0
-      this.sliderWidth = null
-    } else {
-      this.sliderOffsetX = tab.$el.offsetLeft
-      this.sliderWidth = tab.$el.clientWidth
-      this.sliderOffsetY = 0
-      this.sliderHeight = null
+    const includeAnimation = () => {
+      if (!animationIncluded.value) {
+        requestAnimationFrame(() => {
+          animationIncluded.value = true
+        })
+      }
+    }
+
+    const redrawTabs = () => {
+      const oldShowPaginationValue = showPagination.value
+
+      updatePagination()
+
+      if (oldShowPaginationValue === showPagination.value) {
+        updateTabsState()
+        includeAnimation()
+      } else {
+        requestAnimationFrame(() => {
+          updateTabsState()
+          includeAnimation()
+        })
+      }
     }
-  }
 
-  resetSliderSizes () {
-    this.sliderWidth = 0
-    this.sliderHeight = 0
-  }
+    const selectTab = (tab: TabComponent) => {
+      if (!tab) { return }
+
+      tabSelected.value = tab.name || tab.id
 
-  updateStartingXPoint () {
-    this.startingXPoint = 0
-    if (!this.showPagination) {
-      return
+      if (props.stateful) {
+        updateTabsState()
+      }
     }
-    if (this.$props.right) {
-      this.startingXPoint = this.tabsRef?.clientWidth - this.containerRef?.clientWidth
-    } else if (this.$props.center) {
-      this.startingXPoint = Math.floor((this.tabsRef?.clientWidth - this.containerRef?.clientWidth) / 2)
+
+    const registerTab = (tab: TabComponent) => {
+      const idx = tabsList.value.push(tab)
+
+      tab.id = tab.name || idx
     }
-  }
 
-  includeAnimation () {
-    if (!this.animationIncluded) {
-      requestAnimationFrame(() => {
-        this.animationIncluded = true
+    const unregisterTab = (tab: TabComponent) => {
+      tabsList.value = tabsList.value.filter((filteredTab: TabComponent) => filteredTab.id !== tab.id)
+
+      tabsList.value.forEach((tab: TabComponent, idx: number) => {
+        tab.id = tab.name || idx
       })
     }
-  }
 
-  redrawTabs () {
-    const oldShowPaginationValue = this.showPagination
-    this.updatePagination()
-    if (oldShowPaginationValue === this.showPagination) {
-      this.updateTabsState()
-      this.includeAnimation()
-    } else {
+    provide(TabsViewKey, {
+      parentDisabled: props.disabled,
+      tabsList,
+      selectTab,
+      moveToTab,
+      registerTab,
+      unregisterTab,
+    })
+
+    // Lifecycle hooks
+    watch(() => props.modelValue, () => {
+      updateTabsState()
+    })
+
+    onMounted(() => {
+      redrawTabs()
+
       requestAnimationFrame(() => {
-        this.updateTabsState()
-        this.includeAnimation()
+        resizeObserver.value = new ResizeObserver(redrawTabs)
```

Comment:
I think you need to move `new ResizeObserver` outside requestAnimationFrame... Also, ResizeObserver would be nice to move into separate hook, handle here if element is not null and observe it's size.